palm/plugins/dbt/dbt_containerizer.py

```diff
@@ -0,0 +1,92 @@
+from pathlib import Path
+from typing import Optional
+from palm.containerizer import PythonContainerizer
+import sys
+from palm.palm_exceptions import AbortPalm
+import click
+
+class DbtContainerizer(PythonContainerizer):
+    """
+    Containerizer for dbt projects.
+    """
+    def __init__(self, ctx, template_dir: Path, dbt_version: Optional[str] = '0.21.0') -> None:
+        self.ctx = ctx
+        self.project_name = ctx.obj.palm.image_name
+        self.template_dir = template_dir
+        self.dbt_version = dbt_version
+        self.package_manager = ''
+        self.profiles_file = ''
+    
+    def run(self) -> None:
+        """Runs the containerizer.
+
+        Returns:
+            None
+        """
+        super().check_setup()
+        self.package_manager = super().detect_package_manager()
+        self.profiles_file = self.detect_profiles_file()
+        super().generate(self.target_dir, self.replacements)
+        
+
+    def detect_profiles_file(self) -> str:
```

Comment:
style: I'd keep this method and the related `has_profiles_file` `optionally_add_profiles_file` together, makes the code easier to read, rather than having to jump past `replacements` and `validate_python_version` to read this full proceedure

Reply:
fair point - i fixed this!